controllers/certificate_request_controller.go

```diff
@@ -197,17 +197,19 @@ func (r *CertificateRequestReconciler) Reconcile(ctx context.Context, req ctrl.R
 		l.Info("Signing", "CertificateRequest", req)
 		cert, err := ca.Sign(cr.Spec.Request, keyUsage, extKeyUsage)
 		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to sign CSR for signing: %v", err)
+			return ctrl.Result{}, fmt.Errorf("failed to sign CR for signing: %v", err)
 		}
 
 		patch := client.MergeFrom(cr.DeepCopy())
 		cr.Status.Certificate = tlsutil.EncodeCert(cert)
+		setReadyCondition(cmmeta.ConditionTrue, cmapi.CertificateRequestReasonIssued, "Signed")
```

Comment:
Is it so that this only sets the value on the CR but doesn't update it to the API server? How did it work before this change?

Reply:
The status condition is used to update via defer function call. Now I removed the extra path call.